File: app/src/main/java/com/pavelprymak/propodcast/presentation/viewModels/LanguageViewModel.java
package com.pavelprymak.propodcast.presentation.viewModels;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.pavelprymak.propodcast.utils.ServerMocksUtil;

import java.io.IOException;
import java.util.List;

public class LanguageViewModel extends ViewModel {
    private MutableLiveData<List<String>> mLanguagesData = new MutableLiveData<>();

    public LiveData<List<String>> getLanguages() {
        if (mLanguagesData.getValue() == null) {
            try {
                mLanguagesData.setValue((ServerMocksUtil.getLanguagesMock().getLanguages()));
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
        return mLanguagesData;
    }
}

File: app/src/main/java/com/pavelprymak/propodcast/presentation/viewModels/LanguageViewModel.kt
package com.pavelprymak.propodcast.presentation.viewModels

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import com.pavelprymak.propodcast.utils.ServerMocksUtil

import java.io.IOException

class LanguageViewModel : ViewModel() {
    private val mLanguagesData = MutableLiveData<List<String>>()

    val languages: LiveData<List<String>>
        get() {
            if (mLanguagesData.value == null) {
                try {
                    mLanguagesData.setValue(ServerMocksUtil.getLanguagesMock().languages)
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
            return mLanguagesData
        }
}
